### PR TITLE
ref(arq): Use new scopes API

### PR DESCRIPTION
**Pull request by sentrivana**
---

## General Notes

Thank you for contributing to `sentry-python`!

Please add tests to validate your changes, and lint your code using `tox -e linters`.

Running the test suite on your PR might require maintainer approval. Some tests (AWS Lambda) additionally require a maintainer to add a special label to run and will fail if the label is not present.

#### For maintainers

Sensitive test suites require maintainer review to ensure that tests do not compromise our secrets. This review must be repeated after any code revisions.

Before running sensitive test suites, please carefully check the PR. Then, apply the `Trigger: tests using secrets` label. The label will be removed after any code changes to enforce our policy requiring maintainers to review all code revisions before running sensitive tests.

### CODE REVIEW

File: sentry_sdk/integrations/arq.py [hunk 1]
```diff
@@ -72,12 +71,10 @@ def patch_enqueue_job():
 
     async def _sentry_enqueue_job(self, function, *args, **kwargs):
         # type: (ArqRedis, str, *Any, **Any) -> Optional[Job]
-        hub = Hub.current
-
-        if hub.get_integration(ArqIntegration) is None:
+        if sentry_sdk.get_client().get_integration(ArqIntegration) is None:
             return await old_enqueue_job(self, function, *args, **kwargs)
```
Comment (antonpirker) [inline sentry_sdk/integrations/arq.py:75]: We can use `ensure_integration_enabled_async` here.

File: sentry_sdk/integrations/arq.py [hunk 2]
```diff
@@ -89,12 +86,10 @@ def patch_run_job():
 
     async def _sentry_run_job(self, job_id, score):
         # type: (Worker, str, int) -> None
-        hub = Hub(Hub.current)
-
-        if hub.get_integration(ArqIntegration) is None:
+        if sentry_sdk.get_client().get_integration(ArqIntegration) is None:
             return await old_run_job(self, job_id, score)
```
Comment (antonpirker) [inline sentry_sdk/integrations/arq.py:90]: We can use `ensure_integration_enabled_async` here

File: sentry_sdk/integrations/arq.py [hunk 3]
```diff
@@ -165,11 +160,10 @@ def _wrap_coroutine(name, coroutine):
     # type: (str, WorkerCoroutine) -> WorkerCoroutine
     async def _sentry_coroutine(ctx, *args, **kwargs):
         # type: (Dict[Any, Any], *Any, **Any) -> Any
-        hub = Hub.current
-        if hub.get_integration(ArqIntegration) is None:
+        if sentry_sdk.get_client().get_integration(ArqIntegration) is None:
             return await coroutine(ctx, *args, **kwargs)
 
-        hub.scope.add_event_processor(
+        Scope.get_current_scope().add_event_processor(
```
Comment (szokeasaurusrex) [inline sentry_sdk/integrations/arq.py:166]: Are we sure this event processor belongs on the scope, and not the isolation scope?

Comment (sentrivana) [reply sentry_sdk/integrations/arq.py:166]: This I wasn't sure about. Thinking about it again, I think isolation scope makes more sense. Will change.

Comment (sentrivana) [reply sentry_sdk/integrations/arq.py:166]: There's also another `current_scope` a bit [up](https://github.com/getsentry/sentry-python/pull/2878/files#diff-85f717047dcb658107e9e812c993f2487fd94da0a9db77063bbe39e1cc783a09R134). I think this is what confused me. It feels like both should be isolation scopes. What do you think @szokeasaurusrex @antonpirker?

Comment (antonpirker) [reply sentry_sdk/integrations/arq.py:166]: I think event processors should be on the isolation scope, because they should be valid for the whole request.

Comment (szokeasaurusrex) [reply sentry_sdk/integrations/arq.py:166]: I think what @antonpirker said makes sense

Comment (sentrivana) [reply sentry_sdk/integrations/arq.py:166]: Tried it out now and changing either of them (or both) to work with the isolation scope makes the tests fail. 🙃 Investigating...

Comment (sentrivana) [reply sentry_sdk/integrations/arq.py]: Ok this works: https://github.com/getsentry/sentry-python/pull/2878/commits/9d197a3e6d801cabbcd233d97b5f0dba71420c14
* the current transaction is saved on the current scope, so all scopes in the arq integration that try to update transaction attrs also have to go to the current scope
* the event processor itself is now on the isolation scope

File: sentry_sdk/integrations/arq.py (hunk 1 above)
Comment (sentrivana) [reply sentry_sdk/integrations/arq.py]: Seems to work in this integration -- changed

**Review by szokeasaurusrex — APPROVED** [review]

**Pull request merged**